snuba/query/parser/functions.py

```diff
@@ -124,6 +134,26 @@ def parse_function(
         )
 
     name, args, alias = function_tuple
+
+    # If the first argument is a simple column name that refers to an array column
+    # (and we are not arrayJoining on that column, which would make it scalar again)
+    # we assume that the user actually means to check if any (or all) items in the
+    # array match the predicate, so we return an `any(x == value for x in array_column)`
+    # type expression. We assume that operators looking for a specific value (IN, =, LIKE)
+    # are looking for rows where any array value matches, and exclusionary operators
+    # (NOT IN, NOT LIKE, !=) are looking for rows where all elements match (eg. all NOT LIKE 'foo').
+    if name in FUNCTION_TO_OPERATOR:
+        columns = dataset.get_abstract_columnset()
```

Comment:
There is code like this which is duplicated here and in the condition parsing code as well. I'm not sure where to put the code to avoid that.